libtransmission/torrent-files.cc

```diff
@@ -267,7 +267,8 @@ void tr_torrent_files::remove(std::string_view parent_in, std::string_view tmpdi
 
     // Make a list of the top-level torrent files & folders
     // because we'll need it below in the 'remove junk' phase
-    auto top_files = std::set<std::string>{};
+    auto const path = tr_pathbuf{ parent, '/', tmpdir_prefix };
+    auto top_files = std::set<std::string>{ std::string{ path } };
```

Comment:
What files are being left over that requires this? And why does the current code not remove them...?

Reply:
I've seen this when downloading and then adding a new torrent where the source files are the same. You remove the old torrent, but the initially downloaded files remain.

Reply:
@ckerr on macOS Transmission downloads to a temporary location:
`~/Library/Application\ Support/Transmission/unfinished`

when the download is complete, it is moved to your default location, which in my case is my `Downloads` folder:
<img width="654" alt="Screen Shot 2022-05-31 at 6 22 45 am" src="https://user-images.githubusercontent.com/7323792/171092336-fbb50d8d-1297-4041-a3d1-880bd0abd072.png">

the bug occurs when you download anything which contains a folder, eg:
https://archive.org/details/parmenides-poem-balboa

the torrent downloads just fine, and is moved to my default location when downloading finishes just fine. But the containing folder `parmenides-poem-balboa` is not removed from my `unfinished folder`